src/shogun/mathematics/bessel.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Gil Hoben
+ */
+
+extern "C" {
+#include <shogun/mathematics/zbsubs.h>
+}
+
+#include <cmath>
+#include <cstddef>
+#include <numeric>
+#include <stdexcept>
+
+#include <shogun/io/SGIO.h>
+
+namespace shogun
+{
+	double cyl_bessel_k(double nu, double x)
+	{
+#ifdef _LIBCPP_VERSION
+		double zr = x;
+		double zi = 0;
+		int kode = 1;
+		int n = 1;
+		double cy[2] = {std::numeric_limits<double>::quiet_NaN(),
+		                std::numeric_limits<double>::quiet_NaN()};
+		int nz = 1;
+		int ierr = 0;
+
+		zbesk_(&zr, &zi, &nu, &kode, &n, &cy[0], &cy[1], &nz, &ierr);
+
+		if (ierr)
+		{
+			if (ierr == 3)
+			{
+				error("Large arguments -> precision loss (of at least half "
+				      "machine accuracy)");
+			}
+			else if (ierr == 2)
+			{
+				error("Overflow");
+			}
+			else if (ierr == 4)
+			{
+				error("|x|={} or nu={} too large", zr, nu);
+			}
+			else
+			{
+				error("Unexpected error {} returned from zbesk", ierr);
```

Comment:
These errors are similar to the R package, but I assume that is not an issue?

Reply:
I think if there is any chance that a user will face those. They should be improved a tiny bit

Reply:
It is possible, I never had any issues, but I guess if you're using this kernel on any data something can go wrong..